### PR TITLE
Pipeline refactor

### PR DESCRIPTION
**Pull request by jonmjoyce**
Created a structure for executing pipelines. This takes boilerplate code out of ingest.py and allows the logic to be defined in specific class implementations for each type of model (e.g. NOS, FVCOM, RTOFS). This will allow us to add more pipelines as necessary without reimplementing the execution or testing of each individual one.

**I haven't tested the new ingest.py in production yet.**

Future implementation plans include moving additional logic into the appropriate pipelines, perhaps creating a command object to encapsulate src/dest/region data, and maybe auto-registering pipelines like we can with plugins in xpublish. Integration with Pulumi may be nice too so each pipeline automatically creates its own infrastructure.

This also creates building blocks that can be extended with additional functionality for all pipelines such as logging/monitoring, retries, etc. This may or may not be useful for generating STAC items as well, TBD. That should probably go in its own subsystem.

### CODE REVIEW

File: ingest_tools/pyproject.toml [hunk 1]
```diff
@@ -22,6 +22,12 @@ packages = ["ingest_tools"]
 [tool.setuptools.dynamic]
 dependencies = { file = ["requirements.txt"] }
 
+[tool.pytest.ini_options]
+addopts = [
```
Comment (mpiannucci) [inline ingest_tools/pyproject.toml:26]: I think a typo here

**Review by mpiannucci — COMMENTED** [review]
This is so great. How does this interact with the aggregation layer?

**Comment by mpiannucci** [comment]
Also cc @cheryldmorse

**Comment by mpiannucci** [comment]
We potentially should spin up another stack that tests this (in diff bucket). Cheryl is trying this out with her national water model work

File: cloud_aggregator/ingest/ingest.py [hunk 2]
```diff
@@ -26,18 +21,13 @@ def handler(event, context):
...
-        print(f'No ingest available for key: {key}')
\ No newline at end of file
+    # TODO: These could get auto-registered
+    context.add_pipeline('nos_roms', NOS_Pipeline())
+    context.add_pipeline('fvcom', FVCOM_Pipeline())
+    context.add_pipeline('rtofs', RTOFS_Pipeline())
+
+    matching = context.get_matching_pipelines(key)
+    for pipeline in matching:
+        pipeline.run(context.get_region(), bucket, key, context.get_dest_bucket())
```
Comment (mpiannucci) [inline cloud_aggregator/ingest/ingest.py:33]: There's a decision to make here about whether files can trigger only one job. Currently only one pipeline is triggered and then it exits.

**Comment by jonmjoyce** [comment]
> This is so great. How does this interact with the aggregation layer?

I'm not sure yet but I can look at that next. Just wanted to get this concept out first. 

> We potentially should spin up another stack that tests this (in diff bucket). Cheryl is trying this out with her national water model work

I like that idea too, how hard is it to configure different Pulumi environments? The other part of these changes that I may not have explicitly mentioned is that it should be easier to move dataset-related configurations to config files if desired in the future. We should be able to change src/dest buckets easily as well.

**Comment by mpiannucci** [comment]
> > This is so great. How does this interact with the aggregation layer?
> 
> I'm not sure yet but I can look at that next. Just wanted to get this concept out first.
> 
> > We potentially should spin up another stack that tests this (in diff bucket). Cheryl is trying this out with her national water model work
> 
> I like that idea too, how hard is it to configure different Pulumi environments? The other part of these changes that I may not have explicitly mentioned is that it should be easier to move dataset-related configurations to config files if desired in the future. We should be able to change src/dest buckets easily as well.

Cheryl is working on this currently becuase she is doing her NWM work in a different stack with different buckets and everything not to break what we have

**Review by mpiannucci — APPROVED** [review]

**Pull request merged**